### PR TITLE
ME-41 - Updating Base module with ME-36 and Virutal product address i…

### PR DESCRIPTION
**Pull request by dhiraj-toptal**
Updating Base module with ME-36, Virutal product address issue and fixed Magento coding standard bugs

### CODE REVIEW

**Comment by dhiraj-toptal** [comment]
@ynaxonBalance Can you review the PR?

File: Model/AbstractRequest.php [hunk 1]
```diff
@@ -244,33 +272,39 @@ protected function getLinesParams(Quote $quote, $totalShippingAmount = 0)
...
+     * @param Quote $quote
      * @return array
      */
     protected function getBillingAddressParams(Quote $quote)
     {
         $params = [];
 
         if (($billing = $quote->getBillingAddress()) !== null) {
-            $address = (array) $billing->getStreet();
+            $RegionName = $billing->getRegion();
```
Comment (ynaxonBalance) [inline Model/AbstractRequest.php]: format variable name to camelCase

File: Model/AbstractRequest.php [hunk 2]
```diff
@@ -244,33 +272,39 @@ protected function getLinesParams(Quote $quote, $totalShippingAmount = 0)
...
+            $address = (array)$billing->getStreet();
+            if (empty($address[0])) {
+                $billing = $this->accountManagement->getDefaultBillingAddress($quote->getCustomerId());
+                $address = (array)$billing->getStreet();
+                $region = $this->region->create()->load($billing->getRegionId());
+                $RegionName = $region->getName() ?? '';
```
Comment (ynaxonBalance) [inline Model/AbstractRequest.php]: format variable name to camelCase

File: Model/AbstractRequest.php [hunk 3]
```diff
@@ -244,33 +272,39 @@ protected function getLinesParams(Quote $quote, $totalShippingAmount = 0)
...
                 'firstName' => $billing->getFirstname(),
                 'lastName' => $billing->getLastname(),
-                'addressLine1' => (string) array_shift($address),
-                'addressLine2' => (string) implode(' ', $address),
+                'addressLine1' => (string)array_shift($address),
+                'addressLine2' => (string)implode(' ', $address),
                 'zipCode' => $billing->getPostcode(),
                 'countryCode' => $billing->getCountryId(),
-                'state' => (string) $billing->getRegion(),
+                'state' => (string)$RegionName,
```
Comment (ynaxonBalance) [inline Model/AbstractRequest.php]: format variable name to camelCase

File: Model/AbstractRequest.php [hunk 4]
```diff
@@ -279,43 +313,61 @@ protected function getBillingAddressParams(Quote $quote)
...
+     * @param Quote $quote
      * @return array
      */
     protected function getShippingAddressParams(Quote $quote)
     {
         $params = [];
-
         if (($shipping = $quote->getShippingAddress()) !== null) {
-            $address = (array) $shipping->getStreet();
+            $RegionName = $shipping->getRegion();
```
Comment (ynaxonBalance) [inline Model/AbstractRequest.php]: format variable name to camelCase

File: Model/AbstractRequest.php [hunk 5]
```diff
@@ -279,43 +313,61 @@ protected function getBillingAddressParams(Quote $quote)
...
+            $address = (array)$shipping->getStreet();
+            if (empty($address[0])) {
+                if ($quote->getCustomerId()) {
+                    $shipping = $this->accountManagement->getDefaultShippingAddress($quote->getCustomerId());
+                    $address = (array)$shipping->getStreet();
+                }
+                if (!empty($address[0])) {
+                    $address = (array)$shipping->getStreet() ?? '';
+                    $region = $this->region->create()->load($shipping->getRegionId());
+                    $RegionName = $region->getName() ?? '';
```
Comment (ynaxonBalance) [inline Model/AbstractRequest.php]: format variable name to camelCase

File: Model/AbstractRequest.php [hunk 6]
```diff
@@ -279,43 +313,61 @@ protected function getBillingAddressParams(Quote $quote)
...
+                } else {
+                    if (($billing = $quote->getBillingAddress()) !== null) {
+                        $address = (array)$billing->getStreet();
+                        $region = $this->region->create()->load($billing->getRegionId());
+                        $RegionName = $region->getName() ?? '';
```
Comment (ynaxonBalance) [inline Model/AbstractRequest.php]: format variable name to camelCase

File: Model/AbstractRequest.php [hunk 7]
```diff
@@ -279,43 +313,61 @@ protected function getBillingAddressParams(Quote $quote)
...
                 'firstName' => $shipping->getFirstname(),
                 'lastName' => $shipping->getLastname(),
-                'addressLine1' => (string) array_shift($address),
-                'addressLine2' => (string) implode(' ', $address),
+                'addressLine1' => (string)array_shift($address),
+                'addressLine2' => (string)implode(' ', $address),
                 'zipCode' => $shipping->getPostcode(),
                 'countryCode' => $shipping->getCountryId(),
-                'state' => (string) $shipping->getRegion(),
+                'state' => (string)$RegionName,
```
Comment (ynaxonBalance) [inline Model/AbstractRequest.php]: format variable name to camelCase

**Comment by dhiraj-toptal** [comment]
Hi @ynaxonBalance 
I have fixed PR review comments. Can you approve and merge?

**Review by ynaxonBalance — APPROVED** [review]

**Comment by ynaxonBalance** [comment]
Approved the PR, you can merge :)

**Pull request merged**